update quantization doc: add x86 backend as default backend of server inference (#86794)
pytorch/pytorch: docs/source/quantization.rst

Today, PyTorch supports the following backends for running quantized operators efficiently:

> * x86 CPUs with AVX2 support or higher (without AVX2 some operations have inefficient implementations), via `x86` to apply the optimization of `fbgemm <https://github.com/pytorch/FBGEMM>`_ and `onednn <https://github.com/oneapi-src/oneDNN>`_ (see the details at `RFC <https://github.com/pytorch/pytorch/issues/83888>`_)

Reviewer: ```suggestion
* x86 CPUs with AVX2 support or higher (without AVX2 some operations have inefficient implementations), via `x86` optimized by `fbgemm <https://github.com/pytorch/FBGEMM>`_ and `onednn <https://github.com/oneapi-src/oneDNN>`_ (see the details at `RFC <https://github.com/pytorch/pytorch/issues/83888>`_)
```
